<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>816</width>
    <height>521</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Regular Convertor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="verticalSpacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QTextBrowser" name="textBrowser">
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:12pt;&quot;&gt;Welcome in program Regular Convertor!&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Sans Serif'; font-size:12pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Sans Serif'; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:10pt;&quot;&gt;Please choose &lt;/span&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:600;&quot;&gt;mode and conversion&lt;/span&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:10pt;&quot;&gt; or choose &lt;/span&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:600;&quot;&gt;example&lt;/span&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:10pt;&quot;&gt;.&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Sans Serif'; font-size:9pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:9pt;&quot;&gt;You can also change program language.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>816</width>
     <height>17</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSoubor">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_save"/>
    <addaction name="action_open_file"/>
   </widget>
   <widget class="MyTooltipQMenu" name="menuV_b_r_m_du">
    <property name="title">
     <string>Mode selection</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_check_mode"/>
    <addaction name="action_play_mode"/>
    <addaction name="action_step_mode"/>
   </widget>
   <widget class="QMenu" name="menuP_klady">
    <property name="title">
     <string>Examples</string>
    </property>
    <widget class="QMenu" name="examples_RegExp_FA">
     <property name="title">
      <string>RegExp -&gt; FA</string>
     </property>
     <addaction name="separator"/>
     <addaction name="RE_FA_example0"/>
     <addaction name="RE_FA_example1"/>
     <addaction name="RE_FA_example2"/>
     <addaction name="RE_FA_example3"/>
     <addaction name="RE_FA_example4"/>
     <addaction name="RE_FA_example5"/>
     <addaction name="RE_FA_example6"/>
     <addaction name="RE_FA_example7"/>
     <addaction name="RE_FA_example8"/>
     <addaction name="RE_FA_example9"/>
    </widget>
    <widget class="MyTooltipQMenu" name="menuRemove_epsilon">
     <property name="title">
      <string>Remove epsilon</string>
     </property>
     <addaction name="RemoveEpsilon_example0"/>
     <addaction name="RemoveEpsilon_example1"/>
     <addaction name="RemoveEpsilon_example2"/>
     <addaction name="RemoveEpsilon_example3"/>
     <addaction name="RemoveEpsilon_advanced_example1"/>
     <addaction name="RemoveEpsilon_advanced_example2"/>
     <addaction name="RemoveEpsilon_advanced_example3"/>
     <addaction name="RemoveEpsilon_advanced_example4"/>
    </widget>
    <widget class="MyTooltipQMenu" name="menuDeterminizace_KA">
     <property name="title">
      <string>Determinization of the Finite automata</string>
     </property>
     <addaction name="Determinization_example_1"/>
     <addaction name="Determinization_example_2"/>
     <addaction name="Determinization_example_3"/>
     <addaction name="Determinization_example_4"/>
     <addaction name="Determinization_advanced_example_1"/>
     <addaction name="Determinization_advanced_example_2"/>
     <addaction name="Determinization_advanced_example_3"/>
     <addaction name="Determinization_advanced_example_4"/>
    </widget>
    <widget class="QMenu" name="menuContext_free_grammar_to_PushDown_Automata">
     <property name="title">
      <string>Context-free grammar to PushDown Automata</string>
     </property>
     <addaction name="CfgToPda_Example_1"/>
    </widget>
    <addaction name="examples_RegExp_FA"/>
    <addaction name="menuRemove_epsilon"/>
    <addaction name="menuDeterminizace_KA"/>
    <addaction name="menuContext_free_grammar_to_PushDown_Automata"/>
   </widget>
   <widget class="QMenu" name="menuV_b_r_konverze">
    <property name="title">
     <string>Conversion selection</string>
    </property>
    <addaction name="action_RE_to_FA"/>
    <addaction name="action_RemoveEpsilon"/>
    <addaction name="action_Determinization"/>
    <addaction name="m_actionCFGtoPDA"/>
   </widget>
   <widget class="QMenu" name="menuLanguages">
    <property name="title">
     <string>Languages</string>
    </property>
    <addaction name="actionCzech"/>
    <addaction name="actionEnglish"/>
   </widget>
   <addaction name="menuSoubor"/>
   <addaction name="menuV_b_r_m_du"/>
   <addaction name="menuV_b_r_konverze"/>
   <addaction name="menuP_klady"/>
   <addaction name="menuLanguages"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_RE_to_FA">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Regular expresion to Finite machine</string>
   </property>
   <property name="iconText">
    <string>&quot;&quot;</string>
   </property>
   <property name="toolTip">
    <string>&quot;&quot;</string>
   </property>
  </action>
  <action name="action_RemoveEpsilon">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Remove epsilon rules</string>
   </property>
   <property name="iconText">
    <string notr="true">Odstranění epsilon pravidel</string>
   </property>
   <property name="toolTip">
    <string notr="true">Odstranění epsilon pravidel</string>
   </property>
  </action>
  <action name="action_check_mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Individual work</string>
   </property>
   <property name="iconText">
    <string>Samostatná práce</string>
   </property>
   <property name="toolTip">
    <string>In this mode you are solving exercise individualy.</string>
   </property>
  </action>
  <action name="action_play_mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Algorithm stepping</string>
   </property>
   <property name="toolTip">
    <string>In this mode you can step throw algorithm.</string>
   </property>
  </action>
  <action name="action_step_mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Instant checking</string>
   </property>
   <property name="toolTip">
    <string>In this mode will be your solution periodically checked.</string>
   </property>
  </action>
  <action name="RE_FA_example8">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;((ab)*+c)*&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;ε&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;a&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example3">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;ab&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example5">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;a+b&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example4">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;a*&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example6">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;a+b*&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example7">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;(a+b)*&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example9">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE &quot;(a+b)*(a+c)*&quot;</string>
   </property>
  </action>
  <action name="RE_FA_example0">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>RE Ø</string>
   </property>
  </action>
  <action name="RemoveEpsilon_example0">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 1</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/remove_epsilon_example1.png'/&gt;</string>
   </property>
   <property name="statusTip">
    <string>FA=({0,1,2}, {a,b}, {0ε-&gt;1, 1a-&gt;2, 2b-&gt;3},  0, {2,3})</string>
   </property>
   <property name="whatsThis">
    <string/>
   </property>
  </action>
  <action name="RemoveEpsilon_advanced_example1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 1</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src= ':/main_window/pictures/remove_epsilon_advanced_example1.png' /&gt;</string>
   </property>
  </action>
  <action name="action_Determinization">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Determinization of Finite Automata</string>
   </property>
  </action>
  <action name="Determinization_example_1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 1</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_example1.png'/&gt;</string>
   </property>
  </action>
  <action name="Determinization_advanced_example_1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 1</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_advanced_example1.png' /&gt;</string>
   </property>
  </action>
  <action name="action_save">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_open_file">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="RemoveEpsilon_example1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 2</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/remove_epsilon_example2.png'/&gt;</string>
   </property>
  </action>
  <action name="RemoveEpsilon_example2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 3</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/remove_epsilon_example3.png'/&gt;</string>
   </property>
  </action>
  <action name="RemoveEpsilon_example3">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 4</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/remove_epsilon_example4.png'/&gt;</string>
   </property>
  </action>
  <action name="RemoveEpsilon_advanced_example2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 2</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src= ':/main_window/pictures/remove_epsilon_advanced_example2.png' /&gt;</string>
   </property>
  </action>
  <action name="RemoveEpsilon_advanced_example3">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 3</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src= ':/main_window/pictures/remove_epsilon_advanced_example3.png' /&gt;</string>
   </property>
  </action>
  <action name="RemoveEpsilon_advanced_example4">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 4</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src= ':/main_window/pictures/remove_epsilon_advanced_example4.png' /&gt;</string>
   </property>
  </action>
  <action name="Determinization_example_2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 2</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_example2.png'/&gt;</string>
   </property>
  </action>
  <action name="Determinization_example_3">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 3</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_example3.png'/&gt;</string>
   </property>
  </action>
  <action name="Determinization_example_4">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 4</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_example4.png'/&gt;</string>
   </property>
  </action>
  <action name="Determinization_advanced_example_2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 2</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_advanced_example2.png' /&gt;</string>
   </property>
  </action>
  <action name="Determinization_advanced_example_3">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 3</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_advanced_example3.png' /&gt;</string>
   </property>
  </action>
  <action name="Determinization_advanced_example_4">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced example 4</string>
   </property>
   <property name="toolTip">
    <string>&lt;img src=':/main_window/pictures/determinization_advanced_example4.png' /&gt;</string>
   </property>
  </action>
  <action name="actionCzech">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="pictures.qrc">
     <normaloff>:/main_window/pictures/Flag_of_the_Czech_Republic.png</normaloff>:/main_window/pictures/Flag_of_the_Czech_Republic.png</iconset>
   </property>
   <property name="text">
    <string>Czech</string>
   </property>
  </action>
  <action name="actionEnglish">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="pictures.qrc">
     <normaloff>:/main_window/pictures/Flag_of_the_United_Kingdom.png</normaloff>:/main_window/pictures/Flag_of_the_United_Kingdom.png</iconset>
   </property>
   <property name="text">
    <string>English</string>
   </property>
  </action>
  <action name="m_actionCFGtoPDA">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Context-free grammar to Pushdown Automata</string>
   </property>
  </action>
  <action name="actionTestRefactoredRemoveEpsilon">
   <property name="text">
    <string>TestRefactoredRemoveEpsilon</string>
   </property>
  </action>
  <action name="actionTestRefactoredRegExpToFA">
   <property name="text">
    <string>TestRefactoredRegExpToFA</string>
   </property>
  </action>
  <action name="CfgToPda_Example_1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple example 1</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyTooltipQMenu</class>
   <extends>QMenu</extends>
   <header location="global">mytooltipqmenu.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="pictures.qrc"/>
 </resources>
 <connections/>
</ui>
